src/components/Payment/Payment.tsx

```diff
@@ -1,27 +1,74 @@
-import React from "react";
-import { loadStripe } from "@stripe/stripe-js";
+import React, { useEffect, useState } from "react";
 import { Button } from 'react-bootstrap';
-import api from "../../api/ApiHelper";
+import AbstractPaymentProvider from "../../utils/Payment/AbstractPaymentProvider";
+import availablePaymentProvider from "../../utils/Payment/PaymentUtils";
 
-const stripePromise = loadStripe(
-  "pk_test_51I6N5ZIIRKr1p7dQOGhRRigwIMqgZ3XnoBdbfezFNFgLiR9iaW2YzkRP9kAADCzxSOnqLeqKDVxglDh5uxvY28Dn00vAZR7wQ9"
-);
+let paymentProvider: AbstractPaymentProvider;
 
 function Payment() {
 
-  const onPay = () => {
-    const googleId = localStorage.getItem('googleId');
-    if (googleId) {
-      api.pay(stripePromise, googleId)
-    } else {
-      console.warn('not logged in in google yet')
+  let [productListJsx, _setProductListJsx] = useState<JSX.Element>();
+
+  const setProductsListJsx = (jsx: JSX.Element) => {
+    if (productListJsx !== jsx) {
+      productListJsx = jsx;
+      _setProductListJsx(productListJsx);
+    }
+  }
+
+  const getProducts = async () => {
+    if (!paymentProvider) {
+      paymentProvider = await availablePaymentProvider();
     }
-  };
+    return await paymentProvider.getProducts();
+  }
+
+  const getProductsJsx = async (): Promise<JSX.Element> => {
```

Comment:
Why are you doing it like this?
Just put the products into a state-variable.
Remove async await
Remove col-classNames (very large files)